core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +314,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput' &&
+                  maxSelectionCount === 1) {
+        var numChoices = $scope.getAnswerChoices().length;
+        choiceRules = [];
+        for (var i = 0; i < answerGroups.length; i++) {
+          choiceRules.push(false);
+        }
+
+        var newChoiceRules = function(current) {
+          return choiceRules.map(function(element, index) {
+            if (index !== current || element === true) {
+              return true;
+            } else if (index === current && element !== true) {
+              return false;
+            }
+          });
+        };
+
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              // If the rule_type of the first input is equalTo or containsOne,
+              // then set first choiceRules to true
+              if (ruleSpecs[j].rule_type === 'Equals') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type === 'ContainsAtLeastOneOf') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type ===
+                'DoesNotContainAtLeastOneOf') {
+                // If the ruleType is omits one of, then set the others to true
+                var current = i;
+                choiceRules = newChoiceRules(current);
```

Comment:
OK, I'm kind of confused about what's going on here...

I thought the idea was to not show the default answer group if all possible answers would have been addressed by the other (non-default) answer groups. So, a way to do this is to create an array with one element for each possible answer choice in the interaction itself. Then, go through each of the rules in each of the answer groups, and mark off any answer choices that the rule would take care of.

If, at the end, all answer choices are marked, then you would not show the default answer group. Otherwise, you would.

But this means that the array you use to keep track of things should be based on answerChoices, as opposed to answerGroups. (It might be worth making sure that you're clear about the difference between these -- once you've got the conceptual clarity, the final code should be much simpler.)